### PR TITLE
Added tooltips for all parts of the Create Commons Form except for ending date, as of this commit ending date does not yet exist

### LINKED ISSUE #1: FEATURE: Explanation of fields on Create Commons Page
### Have you updated the title?

- [x] I have updated the issue's title.

### As a...

admin

### I can...

see tooltips to better understand the values on the Create Commons Page

### So that...

I can more effectively set up the game.

### Discussion

There is a way to set "tooltips" on HTML elements, i.e. little pop ups to provide help text.

You can see that if you hover over any of the elements in the drop down menus for the Health Update Formulas on the Create Commons page, these pop up, for example:

image

And: 

image

We want to add some tooltips to the other fields of the Create Commons Page so that the admin better understands the fields on that page.

image

For each of the fields, we might have a tool tip like the ones below:

| Field | Tooltip |
|------|--------|
| Commons Name | This is the name farmers will see when joining the game |
| Starting Balance | Each farmer starts with this amount of money in dollars |
| Cow Price | This is the price to purchase cows.  The selling price is this amount times the health of the cows on that farm |
| Milk Price | This is the amount of money the farmer earns in profits for each cow every time it is milked if it is at 100% health.  When a cow is at health less than 100%, the amount earned is multiplied by that percentage (e.g. 75% of the milk price if the health is at 75% |
| Degradation Rate | This number controls the rate at which cow health decreases when the number of cows in the commons is greater than the effective carrying capacity.  The way in which the number is used depends on the selected Health Update Formulas below. |
| Carrying Capacity |  This is the minimum carrying capacity for the commons; at least this many cows may graze in the commons regardless of the number of players.  If this number is zero, then only the Capacity Per User is used to determine the actual carrying capacity. |
| Capacity Per User | When this number is greater than zero, the commons will be able to support at least this many cows per farmer; that is, the effective carrying capacity of the commons is the value of Carrying Capacity, or Capacity Per User times the number of Farmers, whichever is greater.  If this number is zero, then the Carrying Capacity is fixed regardless of the number of users. |
| Starting Date | This is the starting date of the game; before this  date, the jobs to calculate statistics, milk the cows, and report profits, etc. will not be run on this commons. |
| Ending Date | As of the creation of this issue, there was no ending date, but if one is added (in a separate issue) it should be documented as: This is the ending date of the game (the last day of the game).  After this date, the jobs to calculate statistics, milk the cows, and report profits, etc. will not automatically be run on this commons. |
| Show Leaderboard | When checked, regular users will have access to the leaderboard for this commons.  When unchecked, only admins can see the leaderboard for this commons. |

### Acceptance Criteria

- [x] Each of the tooltips described above is added to the commons page.

### Implementation Todos

This is pretty much entirely within the form for the commons page.

### PR DESCRIPTION
**Pull request by ChinDandekar**
## Overview

In this PR, I added tooltips for all parts of the Create Commons Form except for ending date. This page is only accessible to Admins, and allows any admin creating a new common to see what each field in the form means when they hover their mouse over that field.

## Screenshots
Screenshot 2023-11-17 at 3 48 43 PM
Screenshot 2023-11-17 at 3 48 56 PM
Screenshot 2023-11-17 at 3 49 03 PM
Screenshot 2023-11-17 at 3 49 08 PM
Screenshot 2023-11-17 at 3 49 13 PM
Screenshot 2023-11-17 at 3 49 19 PM
Screenshot 2023-11-17 at 3 49 24 PM
Screenshot 2023-11-17 at 3 49 28 PM
Screenshot 2023-11-17 at 3 49 36 PM

## Feedback Request
Please give suggestions on the following:
- The way this is implemented is the best I can think of. Is it acceptable?

## Validation
1. Download this branch.
2. Run the project backend and frontend locally.
3. Log in as an Admin and go to Admin -> Create Commons.
4. Test this function.

Running on [https://proj-happy-cows-chindandekar-dev.dokku-02.cs.ucsb.edu/](https://proj-happy-cows-chindandekar-dev.dokku-02.cs.ucsb.edu/)

## Linked Issues

Linked with #18 

Closes #1

### CODE REVIEW

File: frontend/package-lock.json
Comment (GlassBricks) [inline frontend/package-lock.json:1]: `package-lock.json` should probably not be included in this PR.
This seems to be part of the last commit; could you revert it (or drop the commit and force-push)?

**Review by GlassBricks — CHANGES_REQUESTED** [review]

File: frontend/package-lock.json
Comment (chriscerie) [reply frontend/package-lock.json:1]: It's much worse to develop with an updated dependency set, commit those changes, then discard the lockfile diff. The team should be working with consistent dependency trees or you risk running into "it worked on my machine" situations, even if they're only patch bumps. The better approach here would be to suggest using `npm ci` for the future to preserve lock versions.

Comment (pconrad) [reply frontend/package-lock.json:1]: Chris, I think the point is that there shouldn't be a commit to package-lock.json unless there has been a change to package.json, which in this case, there has not.  I agree with @GlassBricks : package-lock.json should not be a part of this PR.

Comment (pconrad) [reply frontend/package-lock.json:1]: @ChinDandekar do you know how to do that git magic to get rid of the changes to package-lock.json while keeping your other changes intact?  If so, please do it.  If not, let me know, and I'll walk you through it.

**Review by pconrad — CHANGES_REQUESTED** [review]
Please remove package-lock.json from the PR.

One easy way to do that is this:

```
git checkout chinmay-explain-create-commons-page
git pull origin chinmay-explain-create-commons-page
git fetch
```

That gets you on the branch, and updates your branch plus the local value of where origin/main points.

Then do this: this keeps the files on the file system the same (that's what --soft means here) but resets the pointer of your branch to where main is on Github, so that you have a starting point to rebuild your commit history:

```
git reset --soft origin/main
git status
```

Now you'll see the files that are changed from main, which includes both 
* src/main/components/Commons/CommonsForm.js
* frontend/package-lock.json

Do a new commit but ONLY add the .js file:
```
git add src/main/components/Commons/CommonsForm.js
git commit -m "cd - add tooltips to CommonsForm.js"
```

Then you'll need to do a force push because you did a reset on the branch pointer:

```
git push origin chinmay-explain-create-commons-page -f
```

File: frontend/package-lock.json
Comment (chriscerie) [reply frontend/package-lock.json:1]: @pconrad Whether the lockfile change should be committed or not isn't my point. The lockfile is meaningless if the local development and testing environment deviates from it. If the OP had only reverted the commit and nothing else, their local build would deviate from everyone else's. A better solution is to suggest rebuilding dependencies using `npm ci` so this issue doesn't come up again. With or without reverting that lockfile diff first (and any possible security updates that went with the patch bumps) is aside from my point, but either way simply suggesting to revert the change isn't the answer.

In this case the chances of this mattering is small as it was only 2 patch bumps. But for the sake of better software practices (see axios that shipped a [buggy release](https://github.com/axios/axios/issues/5038) and broke downstream consumers that did not pin dependencies properly), I hope this pattern changes.

Comment (pconrad) [reply frontend/package-lock.json:1]: @chriscerie You raise some interesting points; let's discuss those (you and me) by email.   

In terms of this PR, I think @GlassBricks 's feedback was appropriate.   

@ChinDandekar please just focus on removing `package-lock.json` from the PR (I'm making a guide right now with two ways to do that; I'll link to that when it's done.)   You can disregard the rest of the thread; if something comes out of it from my discussions with @chriscerie in terms of modifying our suggested development process for frontend code,  I'll let the entire class know.

Comment (GlassBricks) [reply frontend/package-lock.json:1]: If I may comment:
My point in asking to revert the change is, while updating packages can be a good idea, that should be done as a separate PR, not tagged along in here.
Perhaps more complete instructions would be to also run `npm install` after reverting the change, so the package versions sync up.

**Review by pconrad — APPROVED** [review]
LGTM

**Pull request merged**